api_app/db/repositories/workspaces.py

```diff
@@ -67,14 +67,18 @@ def create_workspace_item(self, workspace_input: WorkspaceInCreate, auth_info: d
 
         template = self.validate_input_against_template(workspace_input.templateName, workspace_input, ResourceType.Workspace, user_roles)
 
+        # allow for workspace template taking a single address_space or multiple address_spaces
         address_space_param = {"address_space": self.get_address_space_based_on_size(workspace_input.properties)}
+        address_spaces_param = {"address_spaces": [address_space_param]}
```

Comment:
I'm missing the plan here...
You're adding a new field to the database workspace object but it's value is the existing/single address space we have been using.

Reply:
When a workspace service is deployed that needs an address space, an additional address will be added to the `address_spaces` array. The PR will follow this one, but includes:

```
    # if template has address_space get an address space
    if resource_template.properties.get("address_space"):
        # check workspace has address_spaces property
        if not workspace.properties.get("address_spaces"):
            raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=strings.WORKSPACE_DOES_NOT_HAVE_ADDRESS_SPACES_PROPERTY)
        workspace_service.properties["address_space"] = workspace_repo.get_address_space_based_on_size(workspace_service_input.properties)
        workspace_patch = ResourcePatch()
        workspace_patch.properties = {"address_spaces": workspace.properties["address_spaces"] + [workspace_service.properties["address_space"]]}
        workspace_repo.patch_workspace(workspace, workspace_patch, etag, resource_template_repo, user)

```

Reply:
Ok, I understand. 
Just a question to consider for when this extra functionality will be introduced - do you ensure the workspace_service operation wait for the workspace update one? 
(you can resolve this after reading)

Reply:
I'm handling the workspace upgrade in the workspace service pipeline. Let's discuss it in the next PR.